src/lib/mina_graphql/mina_graphql.ml

```diff
@@ -1976,6 +2058,29 @@ module Types = struct
                 Error "Invalid format for token."
           with _ -> Error "Invalid format for token.")
 
+    (* TODO: define a type otherwise identical to Party.Signed.t, but
+       which makes the nonce optional
+    *)
+    let snapp_party_signed_arg : Party.Signed.t option arg_typ =
+      scalar "SnappPartySigned"
+        ~doc:"A party to a Snapp transaction with a signature authorization"
+        ~coerce:(fun party_signed ->
+          let json = to_yojson party_signed in
+          Party.Signed.of_yojson json)
+
+    let snapp_party_arg : Party.t option arg_typ =
+      scalar "SnappParty" ~doc:"A party to a Snapp transaction"
+        ~coerce:(fun party ->
+          let json = to_yojson party in
+          Party.of_yojson json)
```

Comment:
Sorry, can we spell this out explicitly so we can add doc strings to every field?

Reply:
See #9492.